### PR TITLE
Fixed ShutterCalibration<x>

### PR DESCRIPTION
**Pull request by oscarciurana**
## Description:

ShutterCalibration<x> was not working properly. i.e.
In a shutter calibrated using (the numbers were obtained from shutter real position)
    ShutterCalibration1 14900 19500 24100 27850 28600
We expect that when setting
    ShutterPosition1 30
the shutter has to be moved to 14900 real position
    ShutterPosition1 50
to 19500 real position
    ShutterPosition1 70
to 24100 real position
and so on

The previous code was not working this way. I cannot understand the calibration algorithm used in ShutterPercentToRealPosition/ShutterRealToPercentPosition.

With this fix shutter is moved to same positon specified in ShutterCalibration<x> positions, and the intermediate positions are calculated using linear interpolation betwen the previous and the next calibrated positions.

## Checklist:
  - [x] The pull request is done against the latest dev branch
  - [x] Only relevant files were touched
  - [x] Only one feature/fix was added per PR.
  - [x] The code change is tested and works on core 2.6.1
  - [x] The code change pass travis tests. **Your PR cannot be merged unless tests pass**
  - [x] I accept the [CLA](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md#contributor-license-agreement-cla).

### CODE REVIEW

File: tasmota/xdrv_27_shutter.ino [hunk 1]
```diff
@@ -111,21 +116,23 @@ int32_t ShutterPercentToRealPosition(uint32_t percent, uint32_t index)
       if (0 == Settings.shuttercoeff[j][index]) {
         AddLog_P2(LOG_LEVEL_ERROR, PSTR("SHT: RESET/INIT CALIBRATION MATRIX DIV 0"));
         for (uint32_t k = 0; k < 5; k++) {
-          Settings.shuttercoeff[k][index] = SHT_DIV_ROUND(calibrate_pos[k+1] * 1000, calibrate_pos[5]);
+          Settings.shuttercoeff[k][index] = (uint32_t)calibrate_pos[k+1] * 1000 / calibrate_pos[5];
```
Comment (to-scho) [inline tasmota/xdrv_27_shutter.ino]: When not using SHT_DIV_ROUND here we will introduce an old bug again that shutter might not reach the position ask in command ShutterPosition.

**Review by to-scho — CHANGES_REQUESTED** [review]
This commit will break the described shutter calibration procedure and will also break already calibrated shutters.

Idea of calibration is not that Shutterposition1 30 reach same position as in calibration step. Calibration  is not done by steering the shutter to 30% real position and to pass that value in the calibration command. Instead you drive the uncalibrated shutter by Shutterposition1 30 to the 'uncalibrated' 30% and do the real world measurement. After calibration the Shutterposition1 30 will drive the shutter to real world 30% and not to the uncalibrated 30% as before.

Or in other words, ShutterCalibration1 takes values 'inverted' to your thought about calibration. The unit of the measurements given to ShutterCalibration1 is 'cm' and not time "(the numbers were obtained from shutter real position)".

**Comment by to-scho** [comment]
See https://github.com/tasmota/docs/pull/67 for clarification about calibration procedure.

**Comment by oscarciurana** [comment]
Thank you for your clarification in the calibration process. Sorry about my confusion, I'm not a native english speaker and I misunderstood the calibration procedure.

**Pull request closed without merging**